src/git-source-provider.ts

```diff
@@ -0,0 +1,285 @@
+import * as core from '@actions/core';
+import * as coreCommand from '@actions/core/lib/command';
+import * as fs from 'fs';
+import * as fsHelper from './fs-helper';
+import * as gitCommandManager from './git-command-manager';
+import * as io from '@actions/io';
+import * as path from 'path';
+
+let authConfigKey = `http.https://github.com/.extraheader`;
+
+export async function getSource(
+    repositoryPath: string,
+    repositoryOwner: string,
+    repositoryName: string,
+    ref: string,
+    commit: string,
+    clean: boolean,
+    fetchDepth: number,
+    lfs: boolean,
+    accessToken: string) {
+
+    core.info(`Syncing repository: ${repositoryOwner}/${repositoryName}`);
+    let repositoryUrl = `https://github.com/${encodeURIComponent(repositoryOwner)}/${encodeURIComponent(repositoryName)}`;
+
+    // Remove conflicting file path
+    if (fsHelper.fileExistsSync(repositoryPath)) {
+        await io.rmRF(repositoryPath);
+    }
+
+    // Create directory
+    let isNew = false;
+    if (!fsHelper.directoryExistsSync(repositoryPath)) {
+        isNew = true;
+        await io.mkdirP(repositoryPath);
+    }
+
+    // Git command manager
+    core.info(`Working directory is '${repositoryPath}'`);
+    let git = await gitCommandManager.CreateCommandManager(repositoryPath, lfs);
+
+    // Existing directory
+    if (!isNew) {
```

Comment:
I would take this whole if block and make a method since the parent method is really large and complicated. You can make the return value from this be equal to recreate, which lets you short-circuit all the if blocks and directly return from the method

Reply:
+1 it reads better now